@navikt/core/react/src/form/autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { Dispatch, SetStateAction, forwardRef, useState } from "react";
+import { createContext } from "../../util/create-context";
+import { useMergeRefs } from "../../util/hooks";
+import { createDescendantContext } from "../../util/hooks/descendants/useDescendant";
+import { SlottedDivElementRef } from "./SlottedDivElement";
+import { remove_virtual_focus, set_virtual_focus } from "./utils";
+
+// The recipe for success!
+//
+// 1. look at Menu PR, copy pasta + adjust
+// 2. useDescendants for switching focus (iterating focus through list)
+// 3. <Slot /> usage? it's (radix) the enabler for asChild (merge parent props into single child + melt away)
+// 4. <dismissableLayer /> click/focus outside
+// 5. <Floating /> utility
+//
+// perhaps a desired API?
+//
+// <Autocomplete>
+//    <Autocomplete.Anchor>
+//        ...
+//    </Autocomplete.Anchor>
+//    <Autocomplete.Content>
+//        <Autocomplete.Item>
+//           ...
+//        </Autocomplete.Item>
+//        <Autocomplete.Item>
+//           ...
+//        </Autocomplete.Item>
+//    <Autocomplete.Content>
+// </Autocomplete>
+
+export const [
+  AutocompleteDescendantsProvider,
+  useAutocompleteDescendantsContext,
+  useAutocompleteDescendants,
+  useAutocompleteDescendant,
+] = createDescendantContext<SlottedDivElementRef>();
```

Comment:
Ref where to type the additional props
```suggestion
] = createDescendantContext<SlottedDivElementRef, {   handleVirtualOnFocus?: (
    node_to_focus: HTMLElement,
    node_to_blur?: HTMLElement,
  ) => void;
  handlePick?: () => void;
  }>();
```